daemon/access.go

```diff
@@ -189,44 +190,46 @@ func requireInterfaceApiAccessImpl(d *Daemon, r *http.Request, ucred *ucrednet,
 	if err != nil {
 		return Forbidden("internal error: cannot get connections: %s", err)
 	}
+	foundMatchingInterface := false
 	for refStr, connState := range conns {
-		if !connState.Active() || connState.Interface != interfaceName {
+		if !connState.Active() || !strutil.ListContains(interfaceNames, connState.Interface) {
 			continue
 		}
 		connRef, err := interfaces.ParseConnRef(refStr)
 		if err != nil {
 			return Forbidden("internal error: %s", err)
 		}
 		if connRef.PlugRef.Snap == snapName {
-			r.RemoteAddr = ucrednetAttachInterface(r.RemoteAddr, interfaceName)
-			return nil
+			r.RemoteAddr = ucrednetAttachInterface(r.RemoteAddr, connState.Interface)
```

Comment:
I feel that we need a test to check that all snap interfaces are attached to remoteAddr, I think extending `TestRequireInterfaceApiAccessImpl` would be enough.

Reply:
I've modified the test in the most recent commit, please let me know if this is what you had in mind.

Reply:
Perfect